samples/bluetooth/observer/sample.yaml

```diff
@@ -3,20 +3,30 @@ sample:
 tests:
   sample.bluetooth.observer:
     harness: bluetooth
+    tags: bluetooth
     platform_allow:
       - qemu_cortex_m3
       - qemu_x86
       - nrf52840dk/nrf52840
     integration_platforms:
       - qemu_cortex_m3
-    tags: bluetooth
   sample.bluetooth.observer.extended:
     harness: bluetooth
+    tags: bluetooth
     extra_args: CONF_FILE="prj_extended.conf"
     platform_allow:
       - qemu_cortex_m3
       - qemu_x86
       - nrf52840dk/nrf52840
-    tags: bluetooth
     integration_platforms:
       - qemu_cortex_m3
+  sample.bluetooth.observer.extended.bbc_microbit:
+    harness: bluetooth
+    tags: bluetooth
+    extra_args:
+      - CONF_FILE="prj_extended.conf"
+      - EXTRA_CONF_FILE=overlay_bbc_microbit-bt_ll_sw_split.conf
```

Comment:
It's kind of odd to have a board specific overlay, instead of having a file in the `boards` directory, although I kind of understand why, given the limitations of the build system. 

It's also somewhat odd that `overlay_bbc_microbit-bt_ll_sw_split.conf` specifically is used (only with) `prj_extended.conf`, but the former has `# CONFIG_BT_EXT_ADV=y`. 

Since `overlay_bbc_microbit-bt_ll_sw_split.conf` is an overlay, does it need values to include values that are otherwise set by `prj.conf` or `prj_extended.conf`?

Reply:
> It's kind of odd to have a board specific overlay

I am moving away from using board specific overlay for samples (many samples seems to not have board specific overlays) due to possibility of supporting different Controller implementations downstream (though nRF51 will not be supported by Nordic downstream). Also, there are too many boards in Zephyr, and board specific overlay will not scale.

> Since overlay_bbc_microbit-bt_ll_sw_split.conf is an overlay, does it need values to include values that are otherwise set by prj.conf or prj_extended.conf?

in the case of BBC Micro Bit overlay the Scan Data Length is reduced to 192 (from 1650) bytes and rx buffer counts lowered due to RAM restrictions.